lib/parse-statements.js

```diff
@@ -113,59 +113,97 @@ function parseImport(result, atRule) {
   const params = valueParser(atRule.params).nodes
   const stmt = {
     type: "import",
+    uri: "",
+    fullUri: "",
     node: atRule,
     media: [],
     layer: [],
+    supports: [],
   }
 
-  // prettier-ignore
-  if (
-    !params.length ||
-    (
-      params[0].type !== "string" ||
-      !params[0].value
-    ) &&
-    (
-      params[0].type !== "function" ||
-      params[0].value !== "url" ||
-      !params[0].nodes.length ||
-      !params[0].nodes[0].value
-    )
-  ) {
-    return result.warn(`Unable to find uri in '${  atRule.toString()  }'`, {
-      node: atRule,
-    })
-  }
+  for (let i = 0; i < params.length; i++) {
+    const node = params[i]
 
-  if (params[0].type === "string") stmt.uri = params[0].value
-  else stmt.uri = params[0].nodes[0].value
-  stmt.fullUri = stringify(params[0])
+    if (node.type === "space" || node.type === "comment") continue
 
-  let remainder = params
-  if (remainder.length > 2) {
-    if (
-      (remainder[2].type === "word" || remainder[2].type === "function") &&
-      remainder[2].value === "layer"
-    ) {
-      if (remainder[1].type !== "space") {
-        return result.warn("Invalid import layer statement", { node: atRule })
+    if (node.type === "string") {
+      if (stmt.uri) {
+        return result.warn(`Multiple url's in '${atRule.toString()}'`, {
+          node: atRule,
+        })
+      }
+
+      if (!node.value) {
+        return result.warn(`Unable to find uri in '${atRule.toString()}'`, {
+          node: atRule,
+        })
+      }
+
+      stmt.uri = node.value
+      stmt.fullUri = stringify(node)
+      continue
+    }
+
+    if (node.type === "function" && /^url$/i.test(node.value)) {
+      if (stmt.uri) {
+        return result.warn(`Multiple url's in '${atRule.toString()}'`, {
+          node: atRule,
+        })
+      }
+
+      if (!node.nodes?.[0]?.value) {
+        return result.warn(`Unable to find uri in '${atRule.toString()}'`, {
+          node: atRule,
+        })
       }
 
-      if (remainder[2].nodes) {
-        stmt.layer = [stringify(remainder[2].nodes)]
+      stmt.uri = node.nodes[0].value
+      stmt.fullUri = stringify(node)
+      continue
+    }
+
+    if (!stmt.uri) {
+      return result.warn(`Unable to find uri in '${atRule.toString()}'`, {
+        node: atRule,
+      })
+    }
```

Comment:
This code is placed here, and again further below, why?

Reply:
The first one, inside the `for`, is an early abort in case something that can not be a url is found before a url. `@import screen url("foo");`

The second is to make sure that any url was found. `@import screen;`

The first one could be dropped, it's not needed functionally, it's only an early return.

Reply:
Ah, missed that the nesting level was different. Goes to show I need to review this in more detail when less tired...